File: backend/app/models/base_term.py
from typing import Optional, List
from pydantic import BaseModel
from datetime import datetime


class BaseTermStats(BaseModel):
    totalQueries: int = 0
    pendingQueries: int = 0
    completeQueries: int = 0
    errorQueries: int = 0


class BaseTerm(BaseModel):
    id: str
    term: str
    category: Optional[str] = None
    createdAt: str
    userId: str
    stats: BaseTermStats = BaseTermStats()


class NewBaseTerm(BaseModel):
    term: str
    category: Optional[str] = None


class BaseTermResponse(BaseModel):
    baseTerm: BaseTerm


class BaseTermsResponse(BaseModel):
    baseTerms: List[BaseTerm]
    total: int


class BulkGenerateRequest(BaseModel):
    countries: List[str]  # ['CA', 'US'] or ['ALL']
    provinces: List[str]  # ['ON', 'TX'] or ['ALL']
    cities: List[str]     # ['Oakville', 'Austin'] or ['ALL']


class BulkGenerateResponse(BaseModel):
    created: int
    skipped: int
    total: int
    message: str

File: frontend/src/types/baseTerm.ts
export interface BaseTermStats {
  totalQueries: number;
  pendingQueries: number;
  completeQueries: number;
  errorQueries: number;
}

export interface BaseTerm {
  id: string;
  term: string;
  category?: string;
  createdAt: string;
  userId: string;
  stats: BaseTermStats;
}

export interface NewBaseTerm {
  term: string;
  category?: string;
}

export interface BulkGenerateRequest {
  countries: string[];  // ['CA', 'US'] or ['ALL']
  provinces: string[];  // ['ON', 'TX'] or ['ALL']
  cities: string[];     // ['Oakville', 'Austin'] or ['ALL']
}

export interface BulkGenerateResponse {
  created: number;
  skipped: number;
  total: number;
  message: string;
}

export interface QueueStatus {
  pending: number;
  queued: number;
  running: number;
  complete: number;
  error: number;
  paused: boolean;
  estimatedTimeRemaining?: number;
  currentlyProcessing?: string;
}

export interface AddToQueueResponse {
  added: number;
  message: string;
}

export interface QueueControlResponse {
  success: boolean;
  message: string;
}
